### DISCUSSION TITLE
How to get the current page number in "Display all pages" mode?

### DISCUSSION THREAD

**Discussion by kalabalik** [wojtekmaj/react-pdf#1565] [Q&A] [answered]
I would like to persist the page number of a document when the user clicks a button and come back to that page number when the user wants. The latter is easy, but how do I get hold of the current page number in the first place? As I am using the [Display all pages](https://github.com/wojtekmaj/react-pdf/wiki/Recipes#display-all-pages) mode, the user is scrolling the document rather than "turning" individual pages. Is there a way to listen to the event that a new page appears on the screen? onRenderSuccess does not seem to do the trick as all pages of my document are rendered (but not displayed) at app start.

**Comment by wojtekmaj** [comment, marked as answer]
I suppose you could use IntersectionObserver API to get callback whenever a given page appears or disappears from the viewport.

**Reply by kalabalik** [reply]
Okay, I will take a look into this api. Thanks for the pointer!
